c++/test/dsets.cpp

```diff
@@ -1492,9 +1492,6 @@ test_dset()
     catch (Exception &E) {
         test_report(nerrors, H5std_string(" Dataset"));
     }
-
-    // Clean up data file
-    cleanup_dsets();
```

Comment:
No need to explicitly call this here since the testing framework will do this now